documentation/configuration_files/application.md

```diff
@@ -0,0 +1,39 @@
+# application.yml
+
+```yml
+# application name - only lowercase alphanumeric character(s) separated by a single hyphen are allowed
+name:
+
+# configuration for locally running the application
+local:
+  dependencies:
+    <id>: # each dependency needs an id
+      image: # docker image
+      environment-variables: # map of environment variables to pass to this dependency
+      secrets: # array of shell environment variables to pass through to this dependency
+      persist: # array of paths within the docker image you would like saved to docker volumes
+  environment-variables: # map of environment variables to pass to every service  
+  secrets: # array of shell environment variables to pass to every services
+
+# configuration for deployments of the application
+remote:
+  dependencies:
+    <id>: # each dependency needs an id
+      type: # dependency type (exocom or rds)
+      template-config: # map of data to pass to the dependency template
```

Comment:
Could we add some links to the remote dependency readme like you did in the service.yml PR?